edgelet/iotedged/src/lib.rs

```diff
@@ -1478,14 +1482,29 @@ where
 
     let workload = start_workload::<_, _, _, _, M>(
         settings,
-        key_store,
+        &key_store.clone(),
```

Comment:
What does the '&' mean? Why not do the same for cert_manager and workload_config.clone()?

Reply:
Without clone, the program can no longer borrow that variable in future calls.

Reply:
`&key_store.clone()` is redundant with `key_store` (no `&` and no `.clone()`). `&` borrows are already `Copy`. Your code compiles fine with just `key_store` as it was originally. It's a different matter for `cert_manager` and `workload_config` that are being taken by value and thus the first usage (ie `start_workload`) does need a `.clone()`

---

There's also the case of why `start_workload` takes a `&` of the key store when it eventually calls `WorkloadService::new` with it and that one clones it so as to have an owned value. That is a silly thing to do - if the callee needs an owned value, it should take an owned value, rather than taking a borrow and cloning it. That allows the caller to know that there are no hidden copies, and the caller could avoid the unnecessary clone if they don't need the original value any more and can just offer it directly. But anyway, it was already wrong before you change so you don't have to fix that.